### PR TITLE
Add targetallocator securitycontext config

### PR DESCRIPTION
**Pull request by frances-barcelos**
**Description:** 
Adding TargetAllocator securityContext configuration

**Link to tracking Issue:**

**Testing:** 
Test added to manifest files

### CODE REVIEW

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -379,7 +379,7 @@ spec:
                 - --zap-log-level=info
                 - --zap-time-encoding=rfc3339nano
                 - --feature-gates=+operator.autoinstrumentation.go,+operator.autoinstrumentation.nginx
-                image: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.89.0
+                image: ghcr.io/francesl/opentelemetry-operator/opentelemetry-operator:0.89.0-27-g0ad467d
```
Comment (frzifus) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:382]:
```suggestion
                image: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.89.0
```

I think we have to revert this.

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -487,7 +487,7 @@ spec:
   minKubeVersion: 1.23.0
   provider:
     name: OpenTelemetry Community
-  version: 0.89.0
+  version: 0.89.0-27-g0ad467d
```
Comment (frzifus) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:490]:
```suggestion
  version: 0.89.0
```
Same here

File: bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -379,7 +379,7 @@ spec:
                 - --zap-log-level=info
                 - --zap-time-encoding=rfc3339nano
                 - --feature-gates=+operator.autoinstrumentation.go,+operator.autoinstrumentation.nginx
-                image: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.89.0
+                image: ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.89.0-27
```
Comment (swiatekm-sumo) [inline bundle/manifests/opentelemetry-operator.clusterserviceversion.yaml:382]: This looks like an accidental change from the bundle update command, please revert it.

File: apis/v1alpha1/opentelemetrycollector_types.go [hunk 4]
```diff
@@ -327,6 +327,10 @@ type OpenTelemetryTargetAllocator struct {
 	// such as regions, zones, nodes, and other user-defined topology domains
 	// https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
 	// +optional
+	SecurityContext *v1.PodSecurityContext `json:"securityContext,omitempty"`
+	// PodSecurityContext configures the pod security context for the
+	// targetallocator.
+	// +optional
```
Comment (swiatekm-sumo) [inline apis/v1alpha1/opentelemetrycollector_types.go:333]: The comment should be above the field definition. As is, it applies to the `ToplogySpreadConstraints` field. You can see in the generated manifests that `SecurityContext` gets the `TopologySpreadConstraints` definition.

File: internal/manifests/targetallocator/deployment_test.go [hunk 5]
```diff
@@ -67,6 +67,10 @@ var testAffinityValue = &v1.Affinity{
 	},
 }
 
+var testSecurityContextValue = &v1.SecurityContext{
+	RunAsUser: pointer.Int64Ptr(1000),
+}
+
```
Comment (swiatekm-sumo) [inline internal/manifests/targetallocator/deployment_test.go:73]: Is this used anywhere?

File: docs/api.md [hunk 6]
```diff
@@ -18083,6 +18083,13 @@ TargetAllocator indicates a value which determines whether to spawn a target all
           Resources to set on the OpenTelemetryTargetAllocator containers.<br/>
         </td>
         <td>false</td>
+      </tr><tr>
+        <td><b><a href="#opentelemetrycollectorspectargetallocatorsecuritycontext">securityContext</a></b></td>
+        <td>object</td>
+        <td>
+          TopologySpreadConstraints embedded kubernetes pod configuration option, controls how pods are spread across your cluster among failure-domains such as regions, zones, nodes, and other user-defined top<br/>
```
Comment (swiatekm-sumo) [inline docs/api.md:18090]: Could you regenerate the documentation as well? We still have the `TopologySpreadConstraints` description here.

**Comment by frances-barcelos** [comment]
Closing this in favour of [PR2419] (https://github.com/open-telemetry/opentelemetry-operator/pull/2419)

**Pull request closed without merging**